### PR TITLE
Fix to use modern ruby syntax to write code/tests

### PR DESCRIPTION
**Pull request by tagomoris**
@frsyuki Any thoughts?
I'll merge this 2 or 3 days later if there's no strong objections.
- https://github.com/xinminlabs/synvert
- https://github.com/xinminlabs/synvert-snippets

### CODE REVIEW

**Comment by iconara** [comment]
Just as #93 this is so extremely unnecessary. Don't change code just to change code. It makes the git history meaningless. Please don't do this.

**Comment by tagomoris** [comment]
`should` syntax will be disabled in future version of rspec, and msgpack-ruby now depends on `rspec ~> 3.3`.
https://github.com/rspec/rspec-expectations/blob/master/Should.md#using-either-expect-or-should-or-both

`should` syntax is not documented in `rspec-expectations` now, and it also doesn't work with objects which subclasses `BasicObject` (fortunately, we didn't use it for now).

**Comment by tagomoris** [comment]
I'm not sure that I have enough time to fix problems at any day in future when Rspec change the default behavior and it makes all tests to fail.
So I want to change test code while I have some time to do it.

**Comment by iconara** [comment]
There's no need to change as long as you don't update RSpec. Also, the old syntax is not going to go away. It's just not going to be enabled by default. That's a big difference.

**Comment by tagomoris** [comment]
Rspec 3.3 dependency was introduced to use `skip` method to create pending specs.
Will NOT we use any new features in future? Will we reject all contributions with new rspec features or say "Rewrite it with old rspec feature or decline it"?

I want not to do such things.

**Comment by iconara** [comment]
I'm saying update when it's necessary, don't do blanked rewrites of almost all of the test code just because it _might_ be needed some time in the future.

And I would absolutely reject contributions that rewrote all test code just to use a single new feature, especially when it's a tiny one like using `skip` instead of `pending`. I'm not even sure that there's a difference between those.

If you feel very strongly about this go ahead, I think it's unnecessary, but it's not nearly on the same level as #93.

File: spec/cases_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 require 'json'
 
-describe MessagePack do
+RSpec.describe MessagePack do
```
Comment (frsyuki) [inline spec/cases_spec.rb]: Is `Rspec.describe` recommended? I can't find about this syntax at http://betterspecs.org/#describe

**Comment by frsyuki** [comment]
I added above comment but looks good for me overall. Apart from #93 which may be very different case, making code clean is generally good idea, in my opinion.
Then any contributors and committers can bring more with less effort. I don't say this is mandatory but it is good so that users can suppose msgpack-ruby is actively maintained OSS.
@tagomoris, I guessed that your point is that new code shouldn't to use old syntax which is considered obsoleted by rspec team, and new code is welcomed.
Ah, do you think it is not good idea, @iconara?

File: spec/cases_spec.rb (hunk 1 above)
Comment (tagomoris) [inline spec/cases_spec.rb]: I'm not sure about it. It's introduced by `synvert` rule.
It looks to reduce `Kernel` modification dependency as much as possible. Not bad.

Comment (tagomoris) [inline spec/cases_spec.rb]: `rspec-core` README example uses `Rspec.describe` style.
https://github.com/rspec/rspec-core#basic-structure

**Comment by iconara** [comment]
I'm closing this since it's not been merged in close to two years.

**Pull request closed without merging**